### PR TITLE
DM-33779: Implement drp_pipe pipeline unit tests

### PR DESCRIPTION
**Pull request by leeskelvin**

### CODE REVIEW

File: ups/drp_pipe.table [hunk 1]
```diff
@@ -15,6 +15,7 @@ setupRequired(faro)
 setupRequired(analysis_drp)
 setupRequired(analysis_tools)
 setupRequired(drp_tasks)
+setupRequired(ap_association)
```
Comment (arunkannawadi) [inline ups/drp_pipe.table:18]: Where is this package used?

**Review by arunkannawadi — APPROVED** [review]
Some minor questions, looks like a great improvement to keep the pipeline(s) from breaking!

File: ups/drp_pipe.table (hunk 1 above)
Comment (arunkannawadi) [reply ups/drp_pipe.table:18]: Ah, I got the answer to this from the conversation on Jira.

File: pipelines/HSC/DRP-Prod.yaml [hunk 2]
```diff
@@ -113,46 +113,6 @@ subsets:
       These detector-level tasks should not be run with 'tract' or 'patch' as
       part of the data ID expression if all reference catalogs for seeding
       forced photometry cover these detector-level quanta are desired.
-  step5:
```
[the remark this answers is not in the document]
Comment (leeskelvin) [reply pipelines/HSC/DRP-Prod.yaml:116]: The [original commit](https://github.com/lsst/drp_pipe/commit/8a81b84fc74161e8aa8caa11e5c1dc4484c9989e), the genesis for this pipeline, did not contain steps 5 or 6 (on purpose). It did however have an import for `$DRP_PIPE_DIR/pipelines/HSC/DRP-RC2.yaml`. This should instead have been importing `$DRP_PIPE_DIR/ingredients/HSC/DRP.yaml`, as it now does. This initial erroneous import brought in steps 5 and 6, which were then copied wholesale into the current file in the initial attempt at writing this file.

Steps 5 and 6 largely contain difference imaging tasks, which are not intended for use with `DRP-Prod`. The remaining non-DIA tasks are replicated elsewhere in the pipeline - e.g., `consolidateSourceTable` in both step 2d and step 6.

Having spoken with @yalsayyad about this, we're happy to remove these steps on this ticket. The unit test on this ticket was the test which identified this error, so it seems appropriate to resolve it here too. The alternative would be to set up another ticket and merge that first (to allow the unit tests in this ticket to pass), however, that seems to me like unnecessary extra effort. If you think it cleaner however, I'd be happy to do so.

File: ups/drp_pipe.table (hunk 1 above)
Comment (leeskelvin) [reply ups/drp_pipe.table:18]: For future reference, here: `ingredients/DRP-minimal-calibration.yaml`.

File: pipelines/HSC/DRP-Prod.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (arunkannawadi) [reply pipelines/HSC/DRP-Prod.yaml:116]: Removing it on this ticket should be fine. But please split up the commit so one of them does just the removal.

[the remark this answers is not in the document]
Comment (leeskelvin) [reply pipelines/HSC/DRP-Prod.yaml:116]: Commit has been split into three separate commits, each focusing on a particular feature change.

[the remark this answers is not in the document]
Comment (arunkannawadi) [reply pipelines/HSC/DRP-Prod.yaml:116]: Thanks, Lee! This is looking much better.

**Pull request merged**